mttl/models/modifiers/sparse_utils/utils.py

```diff
@@ -4,25 +4,30 @@
 import numpy as np
 import torch
 import torch.nn.functional as F
+from peft.import_utils import is_bnb_available
 from scipy.sparse import csr_matrix
 from triton.ops.blocksparse.matmul import _matmul
 
+BNB_AVAILABLE = is_bnb_available()
+if BNB_AVAILABLE:
+    import bitsandbytes as bnb
+
 from mttl.logging import logger
 
 try:
     import linear_sd
-except ImportError:
-
-    logger.info(
-        "linear_sd not available, LinearWithSparseDelta will not work. You can install it from https://github.com/AlanAnsell/peft"
+except Exception as e:
+    logger.info(e)
+    raise ImportError(
+        "linear_sd not available. You can install it with 'pip install git+https://github.com/AlanAnsell/peft.git'"
     )
 
 
 try:
     from spops import csr_add, sddmm
-except ImportError:
```

Comment:
why not keep `ImportError` ?

Reply:
I want to exit the code here if "spops is not available" so that users will know how they can fix this.  I add raise ImportError in the line 30.